utils/token.ts

```diff
@@ -0,0 +1,39 @@
+import { AccessToken, Prisma, PrismaClient } from "@prisma/client";
+
+type TokenArgs = Prisma.AccessTokenCreateInput;
+
+const prisma = new PrismaClient();
+
+/**
+ * Creates a new token if there is not token with the "associatedEmail" field.
+ * More on prisma's upsert method here: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upserthttps://www.prisma.io/docs/reference/api-reference/prisma-client-reference#upsert
+ *
+ * @param token {TokenArgs} - Token that has to be created/updated.
+ * */
+const upsertToken = async (data: TokenArgs): Promise<AccessToken | Error> => {
+  try {
+    const upsertData = await prisma.accessToken.upsert({
+      where: {
+        associatedEmail: data.associatedEmail,
```

Comment:
The where clause needs to be changed as user can also add another account's calendar here.